src/store/SingleAnalysisStore.ts

```diff
@@ -0,0 +1,130 @@
+
+        const proteinProcessor = new ProteinProcessor(proteinCommunicator);
+        await proteinProcessor.compute(peptide, equateIl);
+
+        const ncbiCounts = new Map<NcbiId, number>();
+        for (const protein of proteinProcessor.getProteins()) {
+            ncbiCounts.set(protein.organism, 1);
+        }
+
+        ncbiCounts.set(proteinProcessor.getLca(), 1);
+
+        for (const organismId of proteinProcessor.getCommonLineage()) {
+            ncbiCounts.set(organismId, 1);
+        }
+
+        const ncbiOntologyProcessor = new NcbiOntologyProcessor(ncbiCommunicator);
+        const ncbiOntology = await ncbiOntologyProcessor.getOntology(new CountTable<NcbiId>(ncbiCounts));
+
+        const taxaCounts = new Map<NcbiId, number>();
+
+        for (const protein of proteinProcessor.getProteins()) {
+            taxaCounts.set(protein.organism, (taxaCounts.get(protein.organism) || 0) + 1);
+        }
+
+        const taxaCountTable = new CountTable<NcbiId>(taxaCounts);
+
+        const taxaTree = new NcbiTree(taxaCountTable, ncbiOntology);
+
+        const goProteinProcessor = new GoProteinCountTableProcessor(peptide, equateIl, goCommunicator);
+        await goProteinProcessor.compute(proteinProcessor);
+
+        const goOntologyProcessor = new GoOntologyProcessor(goCommunicator);
+        const goOntology = await goOntologyProcessor.getOntology(goProteinProcessor.getCountTable());
+
+        const ecProteinProcessor = new EcProteinCountTableProcessor(peptide, equateIl, ecCommunicator);
+        await ecProteinProcessor.compute(proteinProcessor);
+
+        const ecOntologyProcessor = new EcOntologyProcessor(ecCommunicator);
+        const ecOntology = await ecOntologyProcessor.getOntology(ecProteinProcessor.getCountTable());
+
+        const ecTree = computeEcTree(ecProteinProcessor.getCountTable(), ecOntology);
+
+        const interproProteinProcessor = new InterproProteinCountTableProcessor(peptide, equateIl, interproCommunicator);
+        await interproProteinProcessor.compute(proteinProcessor);
+
+        const interproOntologyProcessor = new InterproOntologyProcessor(interproCommunicator);
+        const interproOntology = await interproOntologyProcessor.getOntology(interproProteinProcessor.getCountTable());
+
+        assay.value.peptideData = pept2Data.get(peptide)!;
+        assay.value.ncbiOntology = ncbiOntology;
+        assay.value.proteinProcessor = proteinProcessor;
+        assay.value.goProteinCountTableProcessor = goProteinProcessor;
+        assay.value.goOntology = goOntology;
+        assay.value.ecProteinCountTableProcessor = ecProteinProcessor;
+        assay.value.ecOntology = ecOntology;
+        assay.value.interproProteinCountTableProcessor = interproProteinProcessor;
+        assay.value.interproOntology = interproOntology;
+        assay.value.taxaTree = taxaTree;
+        assay.value.ecTree = ecTree;
```

Comment:
Aren't there any performance problems here too?

Reply:
Since these values are not constantly changed (because we don't support filtering for the single peptide analysis), this doesn't seem to be an issue. I guess marking them as `raw`, however, wouldn't cause any harm, but we might better do this in a future release.